### ISSUE TITLE
Vsode snippets  choice bug

### ISSUE THREAD

**Issue by ZKBL** [VSCodeVim/Vim#5538]
**Describe the bug**

Using vscode snippets . When there are two variables that both have choice, only the first one can be selected with tab .

**To Reproduce**
Customize a snipptes like the following
```
	"testsnippet": {
		"prefix": "test",
		"body": ["console.log(${1|:one,two,three|}); ${2|one,two,three|};${3:test}"],
		"description": "test snippet"
	}
```

Then enter this test code snippet into the file, and then, for the first variable you can select choice, but not for the second. If I turn off the vim plugin, it works correctly.

**Screenshots**

![2020-10-30 17-53-18_1](https://user-images.githubusercontent.com/18417140/97694869-29b93e80-1ade-11eb-82dd-a8b699fc198d.gif)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version:**1.17.1**
- VSCode version: **1.50.1**
- OS: **Windows 10 19041.572**

**Comment by berknam** [comment]
This should be fixed with #5540

**Comment by trueinviso** [comment]
I am having this same problem with this snippet that works in normal VSCode just fine:

```
"New service class": {
		"prefix": "nsc",
		"body": [
			"class $3",
			"\tdef initialize($1)",
			"\t\t${1/([^,]+)((,\\s*)|)/@$1 = $1${2:+\n\t\t}/g}",
			"\tend",
			"",
			"\tdef call",
			"\t\t$0",
			"\tend",
			"",
			"\tprivate",
			"",
			"\tattr_reader ${1/([^,]+)((,\\s*)|)/:$1${2:+, }/g}",
			"end"
		],
		"description": "New service class"
	}
```
I guess it doesn't like the Regex that's doing a search and replace on a dynamic group.
